### ISSUE TITLE
Invalid highlighting when link title contain "http" or "https"

### ISSUE THREAD

**Issue by jbleuzen** [preservim/vim-markdown#343]
Hi,

I often write doc with links are like this : 

```
[https://domain.tld](https://domain.tld) not_a_link
```

The thing is that the highlight doesn't end at the bracket. 
I tried to write a test that fails, but not sure I succeed.

```
Given markdown;
[https://domain.tld](https://domain.com) not_a_link

Execute (link with title):
  AssertEqual SyntaxOf('https://domain.tld'), 'mkdLink'
  AssertEqual SyntaxOf('https://domain.com'), 'mkdUrl'
  AssertEqual SyntaxOf('not_a_link'), 'normal'
```

There is one line I my have identified as the culprit but my technical knowledge of vim syntax internal reach its limits and I'm not sure that this line breaks. It's line 66 in `/syntax/markdown.vim`

```
execute 'syn region mkdLink matchgroup=mkdDelimiter  start="\\\@<!!\?\[\ze[^]\n]*\n\?[^]\n]*\][[(]" end="\]" contains=@mkdNonListItem,@Spell nextgroup=mkdURL,mkdID skipwhite' . s:concealends
```

Funny thing is that if I remove the protocol it works...

I reproduce this, when I only load vim-markdown with an empty vimrc. I load vim-markdown with vim-plug.

Hope it helps to understand my issue,
Thanks

**Comment by simon50** [comment]
I also would like to be able to write links this way. It's really annoying that the highlight doesn't stop. It will also continue to the rest of the lines in the same paragraph.

I'll stick to the Pandoc markdown way of writing these kinds of links as `<https://domain.tld>` for now.

**Comment by VVVFO** [comment]
I'm experiencing this issue as well.

**Comment by edubxb** [comment]
Same problem here.

**Comment by BenSYZ** [comment]
Hey guys. I tried to comment out line 73 in`~/.config/nvim/plugged/vim-markdown/syntax/markdown.vim`, which is
```
syn match   mkdInlineURL /https\?:\/\/\(\w\+\(:\w\+\)\?@\)\?\([A-Za-z0-9][-_0-9A-Za-z]*\.\)\{1,}\(\w\{2,}\.\?\)\{1,}\(:[0-9]\{1,5}\)\?\S*/
```

However, it may be a bad method, for which it will not underline the link in `()`, but text in `[]`.

**Comment by lggruspe** [comment]
This seems to have been fixed by #451.
